### ISSUE TITLE
Error in log using LTI Advantage when some fields are empty

### ISSUE THREAD

**Issue by Alex-Jordan** [openwebwork/webwork2#2182]
I have a test user in a course where I am trying to use LTI 1.3. This user did not have a `student_id`, `section`, or `recitation`. After submitting an answer (triggering a grade update) there is this message in the log:

```
Use of uninitialized value in string ne at /opt/webwork/webwork2/lib/WeBWorK/Authen/LTIAdvantage.pm
```

I filled in the three missing fields and now I do not get this in the log. It seems that maybe something should be initialized as the empty string, but it remains undefined at this comparison. Note that I only saw one instance of the message at a time, not three, so maybe it's only one of those three fields where this is happening. I can't easily undo setting the values to test it again for each one.

**Comment by drgrice1** [comment]
It didn't have a line number with it in the log?

The only place where it seems that could happen is on line 708 of LTIAdvantage.pm.  Although it shouldn't be happening there.  It is comparing the user columns from the database to the information that comes from the LMS.  The values from the database does not return undefined values (this goes through line 218 of NewSQL.pm which converts all undefined values to the empty string).  The information from the LMS all is defined as `value // ""`, and so it should also all be defined.

That code is also the same as the code in LTIAdvanced.pm, and I don't think that has happened with that.

I just noticed there is an error in that code that might be related though.  You don't happen to have a `$LTI{v1p1}{modify_user}` method defined in `authen_LIT_1_1.conf` do you?  The error in the LTIAdvantage.pm code is that it is calling that instead of the `$LTI{v1p3}{modify_user}` method.  So if you have the `v1p1` method defined and it is doing something with one of those (section, recitation, or student_id), then it may be setting one of those to undefined (probably because the LTI 1.1 values it might be looking for are not defined for LTI 1.3).

**Comment by Alex-Jordan** [comment]
Sorry, now sure what I copied to paste there, but here is the whole thing:
```
[2023-08-17 10:37:48.21869] [235973] [warn] [MtND_MUTsXZy] [/webwork2/ltiadvantage/launch] Use of uninitialized value in string ne at /opt/webwork/webwork2/lib/WeBWorK/Authen/LTIAdvantage.pm line 709.
```

**Comment by Alex-Jordan** [comment]
>  You don't happen to have a $LTI{v1p1}{modify_user} method defined in authen_LIT_1_1.conf do you?

We do! It's this:
```
$LTI{v1p1}{modify_user} = sub {
  # The self object from LTIAdvanced.pm
  my $self = shift;
  # The user object to be modified
  my $user = shift;

  # Parse context_id for additional information.  E.G.
  #my @course_label=split /\./, $self -> {"context_label"};
  $user->{"section"} = (split /\./, $self->{'section'})[0];

};
```

It lets the WW section be populated with the right thing from D2L (at least, right in our situation). By default we get nothing, and last time I looked up what it was trying to population the WW section with, I can't recall details, but it seemed like the wrong thing.

**Comment by drgrice1** [comment]
So then this is fixed by #2183 and #2184.

**Comment by Alex-Jordan** [comment]
So this explains why I only saw one instance, because `section` was one of the three empty fields.

**Comment by Alex-Jordan** [comment]
Fixed by #2183/#2184.
